tarpc/src/json_transport/mod.rs

```diff
@@ -178,7 +172,7 @@ where
     type Item = io::Result<Transport<TcpStream, Item, SinkItem>>;
 
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        let next = ready!(self.project().incoming.poll_next(cx)?);
+        let next = ready!(Pin::new(&mut self.project().listener.incoming()).poll_next(cx)?);
```

Comment:
This constructs a new incoming stream for every call to `poll_next` -- that doesn't seem right. I understand that the types have been made private, so we can't name then in the struct, but is this really the recommended workaround?